Mathlib/SetTheory/Cardinal/Cofinality.lean

```diff
@@ -1212,6 +1212,12 @@ lemma sup_sequence_lt_omega1 {α} [Countable α] (o : α → Ordinal) (ho : ∀
   rw [Cardinal.isRegular_aleph_one.cof_eq]
   exact lt_of_le_of_lt mk_le_aleph0 aleph0_lt_aleph_one
 
+theorem aleph0_le_cof_of_not_isRegular {o : Ordinal} (h : ¬(aleph o).IsRegular) : ℵ₀ ≤ o.cof := by
```

Comment:
Can this take `IsLimit o` as an assumption instead?

Reply:
I can make that as well, but my use case is working a lot with singular cardinals so I would like to keep this as well. I'll add `¬(aleph o).IsRegular -> o.IsLimit` too. I know `¬κ.IsRegular` is ugly but it's inevitable I'll use it a lot unless we define `IsSingular`.

Reply:
Oh and `Ordinal.aleph0_le_cof : ℵ₀ ≤ o.cof ↔ o.IsLimit` already exists.